File: logger.py
import time
from rich.console import Console


class Logger:
    def __init__(self):
        self.console = Console()
        self.start_time = time.time()

    def log(self, message: str):
        self.console.print(message)

    def print_with_timestamp(self, message: str, style: str = "dim"):
        self.console.print(f"[dim]{self.get_timestamp()}[/dim] {message}")

    def get_timestamp(self) -> str:
        """Get relative timestamp since start"""
        elapsed = time.time() - self.start_time
        minutes = int(elapsed // 60)
        seconds = int(elapsed % 60)
        return f"{minutes:02d}:{seconds:02d}"

    def info(self, message: str) -> None:
        """Print info message with pink styling"""
        timestamp = self.get_timestamp()
        self.console.print(f"[dim]{timestamp}[/dim] {message}", style="bold magenta")

    def error(self, message: str) -> None:
        """Print error message with red styling"""
        timestamp = self.get_timestamp()
        error_console = Console(stderr=True)
        error_console.print(f"[dim]{timestamp}[/dim] {message}", style="bold red")

File: pkg/logger/logger.go
package logger

import (
	"fmt"
	"io"
	"os"
	"time"

	"github.com/fatih/color"
)

// Logger provides timestamped logging functionality
type Logger struct {
	startTime time.Time
	stdout    io.Writer
	stderr    io.Writer
}

// New creates a new Logger instance
func New() *Logger {
	return &Logger{
		startTime: time.Now(),
		stdout:    os.Stdout,
		stderr:    os.Stderr,
	}
}

// getTimestamp returns the elapsed time since logger creation in MM:SS format
func (l *Logger) getTimestamp() string {
	elapsed := time.Since(l.startTime)
	minutes := int(elapsed.Minutes())
	seconds := int(elapsed.Seconds()) % 60
	return fmt.Sprintf("%02d:%02d", minutes, seconds)
}

// PrintWithTimestamp prints a message with a timestamp prefix
func (l *Logger) PrintWithTimestamp(message string) {
	timestamp := color.GreenString(l.getTimestamp())
	fmt.Fprintf(l.stdout, "%s %s\n", timestamp, message)
}

// Info prints an info message with magenta styling
func (l *Logger) Info(message string) {
	timestamp := color.GreenString(l.getTimestamp())
	info := color.New(color.FgMagenta, color.Bold).Sprint(message)
	fmt.Fprintf(l.stdout, "%s %s\n", timestamp, info)
}

// Error prints an error message with red styling to stderr
func (l *Logger) Error(message string) {
	timestamp := color.GreenString(l.getTimestamp())
	errMsg := color.New(color.FgRed, color.Bold).Sprint(message)
	fmt.Fprintf(l.stderr, "%s %s\n", timestamp, errMsg)
}

// Success prints a success message with green styling
func (l *Logger) Success(message string) {
	timestamp := color.GreenString(l.getTimestamp())
	success := color.New(color.FgGreen, color.Bold).Sprint(message)
	fmt.Fprintf(l.stdout, "%s %s\n", timestamp, success)
}

// Warning prints a warning message with yellow styling
func (l *Logger) Warning(message string) {
	timestamp := color.GreenString(l.getTimestamp())
	warning := color.New(color.FgYellow, color.Bold).Sprint(message)
	fmt.Fprintf(l.stdout, "%s %s\n", timestamp, warning)
}

// Dim prints a dimmed message
func (l *Logger) Dim(message string) {
	timestamp := color.GreenString(l.getTimestamp())
	dim := color.New(color.Faint).Sprint(message)
	fmt.Fprintf(l.stdout, "%s %s\n", timestamp, dim)
}
